arcgis-ios-sdk-samples/Utility Network/Display utility association/DisplayUtilityAssociationViewController.swift

```diff
@@ -0,0 +1,155 @@
+            return
+        }
+        
+        // Check if the current viewpoint has an extent.
+        if let viewpoint = mapView.currentViewpoint(with: .boundingGeometry) {
+            // Get all of the associations in extent of the viewpoint.
+            utilityNetwork.associations(withExtent: viewpoint.targetGeometry.extent) { [weak self] (associations, error) in
+                guard let self = self else { return }
+                if let error = error {
+                    print("Error loading associations: \(error)")
+                } else if let associations = associations {
+                    let graphics: [AGSGraphic] = associations.compactMap { association in
+                        // If it the current association does not exist, add it to the graphics overlay.
+                        let associationGID = association.globalID
+                        guard !self.associationsOverlay.graphics.contains(where: {
+                            ($0 as! AGSGraphic).attributes["GlobalId"] as? UUID == associationGID
+                            }) else {
+                                return nil
+                            }
+                        let symbol: AGSSymbol
+                        switch association.associationType {
+                        case .attachment:
+                            symbol = self.attachmentSymbol
+                        case .connectivity:
+                            symbol = self.connectivitySymbol
+                        default:
+                            return nil
+                        }
+                        return AGSGraphic(geometry: association.geometry, symbol: symbol, attributes: ["GlobalId": associationGID, "AssociationType": association.associationType])
+                    }
+                    self.associationsOverlay.graphics.addObjects(from: graphics)
+                }
+            }
+        }
+    }
+    
+    // Populate the legend.
+    func createSwatches() {
+       let swatchGroup = DispatchGroup()
+       var attachmentImage: UIImage?
+       var connectivityImage: UIImage?
+       swatchGroup.enter()
+       attachmentSymbol.createSwatch(withBackgroundColor: nil, screen: .main) { (image, error) in
+           defer { swatchGroup.leave() }
+           if let error = error {
+               print("Error creating swatch: \(error)")
+           } else if let image = image {
+               attachmentImage = image.withRenderingMode(.alwaysOriginal)
+           }
+       }
+       swatchGroup.enter()
+       connectivitySymbol.createSwatch(withBackgroundColor: nil, screen: .main) { (image, error) in
+           defer { swatchGroup.leave() }
+           if let error = error {
+               print("Error creating swatch: \(error)")
+           } else if let image = image {
+               connectivityImage = image.withRenderingMode(.alwaysOriginal)
+           }
+       }
+       swatchGroup.notify(queue: .main) {
```

Comment:
This is an asynchronous operation, so the closure needs to weakly capture `self`.